### ISSUE TITLE
Implement method to disable deletion check

### ISSUE THREAD

**Issue by ultrono** [f9webltd/laravel-deletable#9]
Implement method to disable deletion checks, i.e. so the result of `isDeletable()` is always true and any checks within `isDeletable()` are skipped. The API would look something along the lines of: the following.

A user may have a "deletable" restriction:

```php
namespace App;

use F9Web\LaravelDeletable\Traits\RestrictsDeletion;
use Illuminate\Database\Eloquent\Model;

class User extends Model
{
  use RestrictsDeletion;
  
  public function isDeletable() : bool
  {
    return $this->orders()->doesntExist();
  }  
}
```

Assuming the user had orders, the following would fail and the model would **not** be deleted:

 ```php
$user->delete();
```

The following would allow the deletable check to be skipped and the model **would** be deleted:

 ```php
$user->withoutDeletableRestrictions()->delete()
```

**Comment by BenceSzalai** [comment]
You can do sg like that in userspace:
```php
namespace App;

use F9Web\LaravelDeletable\Traits\RestrictsDeletion;
use Illuminate\Database\Eloquent\Model;

class User extends Model
{
  use RestrictsDeletion;

  protected $allowNextDeletionAttempt = false;
  
  public function isDeletable() : bool
  {
    if ($this->allowNextDeletionAttempt) {
      $this->allowNextDeletionAttempt = false;
      return true;
    }
    return $this->orders()->doesntExist();
  }
  public function withoutDeletableRestrictions() {
      $this->allowNextDeletionAttempt = true;
      return $this;
  }
}
```

Only caveat is that if somewhere `$user->withoutDeletableRestrictions()` is called, but `->delete()` is not chained, the model instance will stay deletable in other parts of the code. This may be acceptable in most cases.

**Comment by ultrono** [comment]
> You can do sg like that in userspace:
> 
> ```
> namespace App;
> 
> use F9Web\LaravelDeletable\Traits\RestrictsDeletion;
> use Illuminate\Database\Eloquent\Model;
> 
> class User extends Model
> {
>   use RestrictsDeletion;
> 
>   protected $allowNextDeletionAttempt = false;
>   
>   public function isDeletable() : bool
>   {
>     if ($this->allowNextDeletionAttempt) {
>       $this->allowNextDeletionAttempt = false;
>       return true;
>     }
>     return $this->orders()->doesntExist();
>   }
>   public function withoutDeletableRestrictions() {
>       $this->allowNextDeletionAttempt = true;
>       return $this;
>   }
> }
> ```
> 
> Only caveat is that if somewhere `$user->withoutDeletableRestrictions()` is called, but `->delete()` is not chained, the model instance will stay deletable in other parts of the code. This may be acceptable in most cases.

Thanks for the reply. Your code would work, but I wanted a nice easy to use API, something like:

```php
$user->withoutDeletableRestrictions()->delete()
```

When I get chance I'll implement this. Would 100% accept a PR.

**Comment by BenceSzalai** [comment]
Yeah, my point was to just put that in a Trait, use in your models or your base model if you have one, and there it is. You can use it like:
 ```php
 $user->withoutDeletableRestrictions()->delete()
 ```

**Comment by ultrono** [comment]
@BenceSzalai Yep see what you mean, the implementation would look similar to that. Again open to PRs as I'm fairly snowed at the moment. 👍
